src/public/app/library/library.component.ts

```diff
@@ -1,41 +1,75 @@
 import { Component, ChangeDetectorRef } from 'angular2/core';
 import roomService from '../room/room.service';
 import library from './library.service';
-import commaizePipe from '../pipes/commaize';
+import { commaizePipe, orderByPipe } from '../pipes';
 import { Song } from '../core/types';
 
 
 let template: string = ` 
- <div style="height:100%; overflow-y: scroll;">
-    <ul class="collection">
-        <li class="collection-item"  style="height: 45px;" *ngFor="#like of songs; #i = index">
-            <div class="row">
-                <div class="col s6" (click)="queue(like)">{{i + 1}}. {{like.title}}</div>
-                <div class="col s2">Plays: {{like.playback_count | commaize }}</div>
-                <div class="col s2">Favs: {{like.favoritings_count | commaize }}</div>
-                <div class="col s2"></div>
-            </div>
-        </li>
-    </ul>
+ <div class="lib-table" style="height:100%; overflow-y: scroll;">
+    <table class=" highlight">
+        <thead>
+          <tr>
+              <th data-field="id" (click)="orderBy('title')">Track</th>
+              <th data-field="id" (click)="orderBy('genre')">Genre</th>
+              <th data-field="name" (click)="orderBy('playback_count')">Plays</th>
+              <th data-field="price" (click)="orderBy('favoritings_count')">Likes</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr *ngFor="#like of songs | orderBy: orderField" (click)="queue(like)">
+            <td>{{ like.title }}</td>
+            <td>{{ like.genre }}</td>
+            <td>{{ like.playback_count | commaize }}</td>
+            <td>{{ like.favoritings_count | commaize }}</td>
+          </tr>
+        </tbody>
+      </table>
 </div>
 `;
 
 @Component({
     template,
     selector: 'library',
-    pipes: [commaizePipe],
+    pipes: [commaizePipe, orderByPipe],
+    styles: [`
+        .lib-table {
+            background: linear-gradient(to left, rgba(85, 189, 151, .8), rgba(0,0,0,.5));
+        }
+        
```

Comment:
agree.

On Tue, Apr 19, 2016 at 5:53 PM, dswen <notifications@github.com> wrote:

> In src/public/app/library/library.component.ts
> <https://github.com/ryanquinn3/beetz/pull/8#discussion_r60336775>:
>
> >  </div>
> >  `;
> >
> >  @Component({
> >      template,
> >      selector: 'library',
> > -    pipes: [commaizePipe],
> > +    pipes: [commaizePipe, orderByPipe],
> > +    styles: [`
> > +        .lib-table {
> > +            background: linear-gradient(to left, rgba(85, 189, 151, .8), rgba(0,0,0,.5));
> > +        }
> > +
>
> hm maybe we should put css in separate files?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ryanquinn3/beetz/pull/8/files/2aabd0b9fad287cfc337421e70eb4c4f70cfdaed#r60336775>
>